components/youtube/new-videos.js

```diff
@@ -0,0 +1,65 @@
+const axios = require('axios')
+const youtube = {
+  type: "app",
+  app: "youtube_data_api",
+}
+
+
+module.exports = {
+  name: "Youtube - New Videos",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    youtube,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+  },
+  methods: {
+    async getVideos(pageToken=null) {
```

Comment:
When you separate the app file, I'd recommend moving this method into the app file so that other sources can benefit from it.

Then you can call the method in this file using `this.youtube.getVideos()` ([see Github example here](https://github.com/PipedreamHQ/pipedream/blob/master/components/github/new-commit.js#L29))

Reply:
Created the app file with the getVideos method in it and am calling it from the component.